components/feature/prompts/src/main/java/mozilla/components/feature/prompts/dialog/LoginDialogFragment.kt

```diff
@@ -210,31 +200,40 @@ internal class LoginDialogFragment : PromptDialogFragment() {
         )
         // The below code is not thread safe, so we ensure that we block here until we are sure that
         // the previous call has been canceled
-        runBlocking { stateUpdate?.cancelAndJoin() }
-        var validateDeferred: Deferred<Result>? = null
-        stateUpdate = scope.launch(IO) {
-            validateDeferred = feature?.loginValidationDelegate?.validateCanPersist(login)
+        runBlocking { validateStateUpdate?.cancelAndJoin() }
+        var validateDeferred: Deferred<LoginValidationDelegate.Result>? = null
+        validateStateUpdate = scope.launch(IO) {
+            validateDeferred = feature?.loginValidationDelegate?.getNewLoginPersistenceAsync(login)
             val result = validateDeferred?.await()
-
             withContext(Main) {
                 when (result) {
-                    is LoginValidationDelegate.Result.CanBeCreated ->
+                    Result.CanBeUpdated -> {
                         setViewState(
+                            confirmButtonEnabled = true,
                             confirmText =
-                            context?.getString(R.string.mozac_feature_prompt_save_confirmation)
+                            context?.getString(R.string.mozac_feature_prompt_update_confirmation)
                         )
-                    is LoginValidationDelegate.Result.CanBeUpdated ->
+                    }
+                    Result.CanBeCreated -> {
                         setViewState(
+                            confirmButtonEnabled = true,
                             confirmText =
-                            context?.getString(R.string.mozac_feature_prompt_update_confirmation)
+                            context?.getString(R.string.mozac_feature_prompt_save_confirmation)
+                        )
+                    }
+                    is Result.Error.Duplicate ->
+                        // If this is an exact duplicate, we will let the user click submit but it
+                        // will be silently eaten when we go to store it
+                        setViewState(
+                            confirmButtonEnabled = true
                         )
-                    is LoginValidationDelegate.Result.Error.EmptyPassword ->
+                    is Result.Error.EmptyPassword ->
                         setViewState(
                             confirmButtonEnabled = false,
                             passwordErrorText =
                             context?.getString(R.string.mozac_feature_prompt_error_empty_password)
                         )
-                    is LoginValidationDelegate.Result.Error.GeckoError -> {
+                    is Result.Error.GeckoError -> {
                         // TODO handle these errors more robustly. See:
                         // https://github.com/mozilla-mobile/fenix/issues/7545
                         setViewState(
```

Comment:
I think I probably should have had this disable the button as well.  Would you mind adding that in?